### PR TITLE
[ios] migrate face detection to mlkit

### LINKED ISSUE #1870: Facedetector is not detecting any faces (iOS)
# Bug Report
- [x] Did you try latest release?
- [x] Did you try master?
- [x] Did you look for existing matching issues?

**Related Modules**
RNCamera
FaceDetector

**Platforms**
iOS

**Versions**
- iOS: 11.x, 12.x (tried on two different devices)
- react-native-camera: master (fc3f7bd)
- react-native: 0.57.3
- react: 16.6

**Description/Current Behaviour**

When I open a frontal camera I get initial callback on onFacesDetected with:
`{type: "face", faces: Array(0), target: 347}`

After that, when I direct camera to face or faces the callback is never called again.

**Expected Behaviour**
Should detect faces and call onFacesDetected callback

**Steps to Reproduce**
Installed Facedetector using CocoaPods with:
` pod 'react-native-camera', subspecs: ['FaceDetector'], path: '../node_modules/react-native-camera'`

My RNCamera settings:

```            type={RNCamera.Constants.Type.front}
            autoFocus
            flashMode={RNCamera.Constants.FlashMode.auto}
            faceDetectionLandmarks={RNCamera.Constants.FaceDetection.Landmarks.all}
            onFacesDetected={this.onFacesDetected}
            onFaceDetectionError={this.onFaceDetectionError}
            faceDetectionMode={RNCamera.Constants.FaceDetection.Mode.accurate}
```

Tried with RNCamera.Constants.FaceDetection.Mode.fast as well and RNCamera.Constants.FaceDetection.Landmarks.none
makes no difference

### LINKED ISSUE #2183: Please make the face detection dependencies optional for iOS
Please make the face detection dependencies optional for iOS

node_modules/react-native-camera/ios/FaceDetector/RNFaceDetectorUtils.h:13:9: 'GoogleMVDataOutput/GoogleMVDataOutput.h' file not found

### PR DESCRIPTION
**Pull request by dov11**
Migrates face detection to MLKit,
Fixes #1870 #2033 #2100
Fixes #2183 - face detection is completely optional now via Cocoapods (which should be the [recommended way](https://facebook.github.io/react-native/blog/#leading-user-problems) now).
Should be fixing  #2184, as I was able to build basic example on XCode 10 after some changes in this PR (was an issue with incomplete RN upgrade on ios side)

### CODE REVIEW

File: ios/RN/FaceDetectorManagerMlkit.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  -(void)setTracking:(id)json queue:(dispatch_queue_t)sessionQueue;
+  -(void)setLandmarksMode:(id)json queue:(dispatch_queue_t)sessionQueue;
+  -(void)setPerformanceMode:(id)json queue:(dispatch_queue_t)sessionQueue;
+  -(void)setClassificationMode:(id)json queue:(dispatch_queue_t)sessionQueue;
+  -(void)findFacesInFrame:(UIImage *)image scaleX:(float)scaleX scaleY:(float)scaleY completed:(postRecognitionBlock)completed;
+  +(NSDictionary *)constants;
+
+  @end
+#else
+  @interface FaceDetectorManagerMlkit : NSObject
```
Comment (jgfidelis) [inline ios/RN/FaceDetectorManagerMlkit.h]: These are all identical, with the exception of the declaration of the enums, why separate into #if and #else?

Comment (dov11) [reply ios/RN/FaceDetectorManagerMlkit.h]: silly development leftover, cleaned up, thanks!

**Comment by sibelius** [comment]
the basic flow is working great, tks

**Pull request merged**